inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -565,14 +565,40 @@ void clDNNEngine::UpdateConfig(CLDNNPlugin::Config& conf, const InferenceEngine:
     }
 }
 
+std::map<std::string, std::string> clDNNEngine::ConvertPerfHintsToConfig(
+        const std::map<std::string, std::string>& network_config,
+        const CLDNNPlugin::Config& plugin_config) const {
+    // deduces the actual settings from the performance hints and returns fully-defined config
+    auto config = network_config;
+    const auto &mode = config.find(PluginConfigParams::KEY_PERFORMANCE_HINT);
+    // the mode may have just arrived to the LoadNetwork, or was set with the plugins' SetConfig
+    if (mode != config.end() || !plugin_config.perfHintsConfig.ovPerfHint.empty()) {
+        const auto mode_name = (mode != config.end())
+                               ? PerfHintsConfig::CheckPerformanceHintValue(mode->second)
+                               : plugin_config.perfHintsConfig.ovPerfHint;
+        //checking streams (to avoid overriding what user might explicitly set in the incoming config or previously via SetConfig)
+        const auto streams = config.find(PluginConfigParams::KEY_GPU_THROUGHPUT_STREAMS);
+        if (streams == config.end() && !streamsSet) {
```

Comment:
Can't this `streamsSet` value be checked right here by `config.find(PluginConfigParams::KEY_GPU_THROUGHPUT_STREAMS) != config.end()`? 
I think it would be more correct, as user may set this flag via LoadExeNetworkImpl config param, not Core::SetConfig call

Reply:
good point, this is copy-paste of the simialr logic from the CPU plugin, which relies on the IStreamsExecutor::Config streamExecutorConfig to store number of streams (and this config has the default value of 1 for #streams, so I need to seperate if this value came from the user, or the default one that we are free to overrride).
If someday the GPU plugin will refactor the code to avoid #streams duplication (via throughput_streams and inside its instance of the IStreamsExecutor::Config) this 'streamsSet' will be needed.
As we discussed before (in the hybrid CPU context) on really generalizing streams executor config per devices, this is desirable refactoring...

Reply:
As a side note notice that CPU plugin doesn't store the number of streams explicitly (as the GPU plugin does with throughput_streams var)  but relies on the IStreamsExecutor::Config streamExecutorConfig to store #streams,

Maybe someday the GPU plugin will refactor the code to avoid #streams duplication (via throughput_streams and inside its instance of the IStreamsExecutor::Config). As we discussed before (in the hybrid CPU context) for really generalizing streams executor config per devices, this is desirable refactoring...